File: js/7-kyu/2019/july/square-every-digit.js
// https://www.codewars.com/kata/square-every-digit/train/javascript

function squareDigits(num){
  let result = []
  const nums = num.toString().split('')
  nums.map((num) => {
    num = num * num
    result.push(num)
  })
  return Number(result.join(''))
}

console.log(squareDigits(9119), 811181);

File: python/2021/september/7-kyu/square-every-digit.py
# https://www.codewars.com/kata/546e2562b03326a88e000020/train/python


def square_digits(num):
    res = ''

    for d in str(num):
        sum = pow(int(d), 2)
        res += str(sum)

    return int(res)


def square_digits(num):
    nums = [int(n) for n in str(num)]
    return int(''.join(str(pow(n, 2)) for n in nums))


def square_digits(num):
    return int(''.join(str(pow(int(digit), 2)) for digit in str(num)))


print(square_digits(9119), 811181)
print(square_digits(0), 0)
